src/lib/time/__init__.js

```diff
@@ -269,25 +391,71 @@ var $builtinmodule = function (name) {
     %Z  Time zone name (no characters if no time zone exists).   
     %%  A literal '%' character.         
     */
-    // accept2dyear
-    mod.accept2year = function accept2year() {
-        throw new Sk.builtin.NotImplementedError("accept2year is not implemented")
-    };
 
-    // strftime()
-    mod.strftime = function strftime() {
-        throw new Sk.builtin.NotImplementedError("strftime is not implemented")
-    };
+    function strftime_f(format, t) {
+        Sk.builtin.pyCheckArgs("strftime", arguments, 1, 2);
+        if (!Sk.builtin.checkString(format)) {
+            throw new Sk.builtin.TypeError("format must be a string");
+        }
+        if (!t)
+        {
+            t = localtime_f();
+        } else if (!(t instanceof struct_time_f)) {
+            t = new struct_time_f(t);
+        } else {
+            // check bounds on given struct_time
+        }
 
-    // strptime()
-    mod.strptime = function strptime() {
-        throw new Sk.builtin.NotImplementedError("strptime is not implemented")
-    };
+        // todo rest of implementation
 
-    // tzset()
-    mod.tzset = function tzset() {
-        throw new Sk.builtin.NotImplementedError("tzset is not implemented")
-    };
+        throw new NotImplementedError("time.strftime() is not yet implemented");
+    }
+
+    mod.strftime = new Sk.builtin.func(strftime_f);
+
+    mod.strftime.__doc__ =
+        "strftime(format[, tuple]) -> string\n" +
+        "\n" +
+        "Convert a time tuple to a string according to a format specification.\n" +
+        "See the library reference manual for formatting codes. When the time tuple\n" +
+        "is not present, current time as returned by localtime() is used.";
+
+    function tzset_f()
+    {
+        Sk.builtin.pyCheckArgs("tzset", arguments, 0, 0);
+
+        throw new NotImplementedError("time.tzset() is not yet implemented");
+    }
+
+    mod.tzset = new Sk.builtin.func(tzset_f);
+
+    mod.tzset.__doc__ =
+        "tzset()\n" +
+        "\n" +
+        "Initialize, or reinitialize, the local timezone to the value stored in\n" +
+        "os.environ['TZ']. The TZ environment variable should be specified in\n" +
+        "standard Unix timezone format as documented in the tzset man page\n" +
+        "(eg. 'US/Eastern', 'Europe/Amsterdam'). Unknown timezones will silently\n" +
+        "fall back to UTC. If the TZ environment variable is not set, the local\n" +
+        "timezone is set to the systems best guess of wallclock time.\n" +
+        "Changing the TZ environment variable without calling tzset *may* change\n" +
+        "the local timezone used by methods such as localtime, but this behaviour\n" +
+        "should not be relied on.";
+
+    function strptime_f()
+    {
+        Sk.builtin.pyCheckArgs("strptime", arguments, 1, 2);   
```

Comment:
Given that this is not yet implemented, this will throw a ```TypeError``` if the wrong number of arguments are passed, rather than a ```NotImplementedError```.  It seems like this should be commented out until the function is implemented.